depot/lib/tram.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Tram do
+  use GenServer
+
+  @moduledoc """
+  Попытка имплементации логики работы трамвая
+  """
+
+  defstruct speed: 0, doors: :closed, accident: false
+
+  def start_link() do
+    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+  end
+
+  def init(_init_arg) do
+    {:ok, %{state: :stop, tram: %Tram{}}}
```

Comment:
А зачем вам _два_ состояния в разных местах? Чем плохо упаковать `state` в `Tram` и использовать тут как:
```suggestion
    {:ok, %Tram{}}
```

Reply:
Блин, точно я не подумал, что так можно было. Одно исправление и туча головной боли ушла с которой я столкнулся. Спасибо.